src/ui/Sort/Sort.ts

```diff
@@ -196,6 +196,108 @@ export class Sort extends Component {
     return _.any(this.options.sortCriteria, (sortCriteria: SortCriteria) => sortId == sortCriteria.toString());
   }
 
+  private findOrCreateRadioGroup() {
+    this.radioGroup = this.findRadioGroup();
+    if (!this.radioGroup) {
+      this.element.setAttribute('role', 'radiogroup');
+      this.radioGroup = this.element;
+    }
+  }
+
+  private createSortButton(innerText?: string) {
+    this.sortButton = $$('span').el;
+    this.sortButton.innerText = this.options.caption || innerText;
+    new AccessibleButton()
+      .withElement(this.sortButton)
+      .withEnterKeyboardAction(() => this.selectAndExecuteQuery())
+      .withArrowsAction((direction, e) => this.onArrowPressed(direction, e))
+      .withLabel(
+        this.canChangeDirection() ? this.getDirectionalLabel(this.initialDirection as VALID_DIRECTION) : this.getOmnidirectionalLabel()
+      )
+      .withRole('radio')
+      .build();
+    this.element.appendChild(this.sortButton);
+  }
+
+  private createDirectionButton() {
+    this.directionButton = $$('span', { className: 'coveo-icon' }, ...this.createIcons()).el;
+    new AccessibleButton()
+      .withElement(this.directionButton)
+      .withSelectAction(e => {
+        e.stopPropagation();
+        this.selectNextCriteriaAndExecuteQuery();
+      })
+      .withArrowsAction((direction, e) => this.onArrowPressed(direction, e))
+      .withLabel(
+        this.getDirectionalLabel(
+          this.initialDirection === VALID_DIRECTION.DESCENDING ? VALID_DIRECTION.ASCENDING : VALID_DIRECTION.DESCENDING
+        )
+      )
+      .withRole('radio')
+      .build();
+    this.element.appendChild(this.directionButton);
+  }
+
+  private onArrowPressed(direction: ArrowDirection, e: Event) {
+    this.selectNextRadioButton(direction === ArrowDirection.RIGHT || direction === ArrowDirection.DOWN ? 1 : -1);
+    e.stopPropagation();
+  }
+
+  private createIcons() {
+    const iconAscending = $$('span', { className: 'coveo-sort-icon-ascending' }, SVGIcons.icons.arrowUp);
+    SVGDom.addClassToSVGInContainer(iconAscending.el, 'coveo-sort-icon-ascending-svg');
+    const iconDescending = $$('span', { className: 'coveo-sort-icon-descending' }, SVGIcons.icons.arrowDown);
+    SVGDom.addClassToSVGInContainer(iconDescending.el, 'coveo-sort-icon-descending-svg');
+    return [iconAscending, iconDescending];
+  }
+
+  private findRadioGroup(element = this.element) {
```

Comment:
Do you need to look up until you end up on the body, or would looking at the direct parent be good enough?

Reply:
Looking at the direct parent could be good *enough*, but it doesn't feel like it would be a too uncommon use-case for clients with their own styling to wrap the individual `Sort` components in something else.

If you still prefer that I make it look at the direct parent only, I can. But by default I'm a bit more tempted to search all ancestors.